### ISSUE TITLE
Maze_Runner Example not working?

### ISSUE THREAD

**Issue by gian1312** [reinforceio/tensorforce#382]
Hi all,
I am testing the maze_runner example. I had to change the states-spec, actions-spec, network-spec to states actions and network. 
Now I am running into:

  File "C:\Users\Gian-Andrea\AppData\Local\conda\conda\envs\tensorflow6\lib\site-packages\tensorforce\core\networks\layer.py", line 499, in tf_apply
    'Invalid input rank for linear layer: {}, must be 2.'.format(util.rank(x))
tensorforce.exception.TensorForceError: Invalid input rank for linear layer: 3, must be 2.

Am I the only one with that problem and did I mess up my installation or is this a known problem with Maze_Runner?

I am quite new to this topic and beg your pardon.

Best Regards
Gian

**Comment by michaelschaarschmidt** [comment]
You need an flatten layer after a discrete layer, consider making a PR for changing the API, as this is a community contributed example

**Comment by gian1312** [comment]
Thank you for this answer.
I will need some time to understand and apply this.
I will let you know if I could resolve this issue.

**Comment by michaelschaarschmidt** [comment]
I fixed the agent_spec stuff. The main problem is just that the embedding layer in discrete environments returns a shape that is not compatible, because we enforce a tensor with shape [batch size, data] in the subsequent layers.

**Comment by gian1312** [comment]
Thank You.
In this case do I need another network? And if yes, is there one network from the examples which is recommended for doing so or do I need to implement one by myself?

**Comment by michaelschaarschmidt** [comment]
I think you can use something like

```
network_spec = [
    dict(type='embedding', indices=100, size=32),
    dict(type'flatten'),
    dict(type='dense', size=32),
    dict(typ
```

I can't run this myself currently but this should work

**Comment by gian1312** [comment]
Thanks for your answer.

I tested it. Sadly, it does just work with the DQN-Agent.
Following specs were used in the .json file in the configs. (I assumed that you allow arbritrary layers in your last line?)
```
[
    {
        "type": "embedding",
        "indices": 100,
        "size": 32
    },

    {	
     	"type": "flatten"
    },

    {
        "type": "dense",
        "size": 32
    }
]
```
For the others, following error occures.
`    'Invalid input rank for linear layer: {}, must be 2.'.format(util.rank(x))
tensorforce.exception.TensorForceError: Invalid input rank for linear layer: 4, must be 2.`

Is it true that for embedded layers the rank is increased by one and that it is just possible using ints as shape?

**Comment by michaelschaarschmidt** [comment]
If you look at the embedding layer you can see how it generates the shape of the output, im just confused why it has 4..I will try to run it

**Comment by michaelschaarschmidt** [comment]
Im seeing  errors even starting the maze runner for some reason, and the environment has not been updated since mid last year (contributed example), so cannot help here sorry

**Comment by gian1312** [comment]
Hi
Thanks for your answer. Beside the problems in the environment, we struggled with the rank issue. Your solution with the flatten layer was and is correct. It didn't work because we did not realize, that there is a baseline which gets initialized by default. The baseline uses a network without a flatten layer by default. I assume, that the TO DO comment in the pg_model.py is obsolet? Nevertheless it seems  to work now.
